pkg/controller/deployment/deployment_controller.go

```diff
@@ -1294,3 +1413,27 @@ func (dc *DeploymentController) rollbackToTemplate(deployment *extensions.Deploy
 	d, err = dc.updateDeploymentAndClearRollbackTo(deployment)
 	return
 }
+
+// isScalingEvent checks whether the provided deployment has been updated with a scaling event
+// by looking at the desired-replicas annotation in the active replica sets of the deployment.
+func (dc *DeploymentController) isScalingEvent(d *extensions.Deployment) bool {
+	newRS, oldRSs, err := dc.getAllReplicaSetsAndSyncRevision(d, false)
+	if err != nil {
+		return false
+	}
+	allRSs := append(oldRSs, newRS)
+	for _, rs := range controller.FilterActiveReplicaSets(allRSs) {
+		if desiredString, ok := rs.Annotations[deploymentutil.DesiredReplicasAnnotation]; ok {
```

Comment:
Do we care about scaling events, or just whether or not we need to adjust the counts on the replica sets to get the desired scale?  The difference is subtle, but AFAICT a "sync" is only loosely couple to actual updates.  If scale and apply / rollout  events both happen before a sync can respond to the first event, we aren't responding to an event, we are bringing the system into spec.  If a user deletes an old replica set manually that owns some of the replicas, we would want to respond to a sync request the same way we would to scaling.